### PR TITLE
Use the offsetalator in cudf::concatenate for strings

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Updates the `cudf::concatenate` logic specialization for strings to use the offsetalator to manage the input and output offsets.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/include/cudf/strings/detail/utilities.hpp [hunk 1]
```diff
@@ -53,6 +53,18 @@ rmm::device_uvector<string_view> create_string_vector_from_column(
...
 
+/**
+ * @brief Return a normalized offset value from a strings offsets column
+ *
+ * @param offsets Input column of type INT32 or INT64
+ * @param index Row value to retrieve
+ * @param stream CUDA stream used for device memory operations and kernel launches
+ * @return Value at `offsets[index]`
+ */
+int64_t get_offset_value(cudf::column_view const offsets,
```
Comment (ttnghia) [inline cpp/include/cudf/strings/detail/utilities.hpp:64]:
```suggestion
int64_t get_offset_value(cudf::column_view const& offsets,
```

File: cpp/src/strings/utilities.cu [hunk 2]
```diff
@@ -128,6 +129,15 @@ special_case_mapping const* get_special_case_mapping_table()
   });
 }
 
+int64_t get_offset_value(cudf::column_view const offsets,
```
Comment (ttnghia) [inline cpp/src/strings/utilities.cu:132]:
```suggestion
int64_t get_offset_value(cudf::column_view const& offsets,
```

File: cpp/src/strings/utilities.cu [hunk 3]
```diff
@@ -128,6 +129,15 @@ special_case_mapping const* get_special_case_mapping_table()
...
+                         size_type index,
+                         rmm::cuda_stream_view stream)
+{
+  return offsets.type().id() == type_id::INT64
```
Comment (ttnghia) [inline cpp/src/strings/utilities.cu:136]: Should we need a type check?
```
CUDF_EXPECTS(offsets.type().id () == type_id::INT64 || offsets.type().id () == type_id::INT32, ....);
```

**Review by ttnghia — APPROVED** [review]

**Review by bdice — APPROVED** [review]
Are tests needed yet, or will that be covered at some later time when we allow 64-bit offsets?

**Comment by davidwendt** [comment]
> Are tests needed yet, or will that be covered at some later time when we allow 64-bit offsets?

All the current tests should cover this. Adding actual 64-bit offsets support will come later.

**Pull request merged**